Add: Azure Active Directory SSO page (#1891)
getsentry/sentry-docs: gatsby/src/docs/accounts/sso/azure-sso.mdx

---
title: 'Azure Active Directory SSO'
sidebar_order: 1
---

## Installation

You'll need an [Office365 developer account](https://developer.microsoft.com/en-us/microsoft-365/dev-program) 

> before installing Azure Active Directory SSO.

Reviewer: This can be deleted. It's for Sentry engineers who need to setup an AzureAD instance.

> 1. Login to the [Azure portal](https://portal.azure.com/) and wait for the Sample Data
> Pack to completly download before continuing.

Reviewer: ```suggestion
1. Login to the [Azure portal](https://portal.azure.com/)
```

> 2. Under Manage Azure Active Directory (the shield picture), click "View", which takes
> you to the Organization Overview page.
> 
> 3. On the left hand navigation, click "Groups". Create a new group, add an owner, and
> members to the group. Set "Group type" to Office 365. For more details about group creation, see the
> [Azure docs](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-groups-create-azure-portal).

Reviewer: ```suggestion
2. You can skip to (3) if you do not need to create a permission group for Sentry
  2.1. Under Manage Azure Active Directory (the shield picture), click "View", which takes
you to the Organization Overview page.

  2.2 On the left hand navigation, click "Groups". Create a new group, add an owner, and
members to the group. Set "Group type" to Office 365. For more details about group creation, see the
[Azure docs](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-groups-create-azure-portal). 
```

4. Return to the Overview page. On the left hand navigation, click "Enterprise Applications".
Then, select "Non-gallery application" and create one for Sentry. Once created, you'll
be directed to "Sentry - Overview."

5. Click on `1. Assign users and groups`, and assign yourself and the group you've created 
to the Sentry app.

6. Click on `2. Setup Single Sign-on`

7. For Section (1) labeled as `Basic SAML2 Configuration`, enter the following data 
in each line.
    

>     Identifier (entity ID): https//[URL of your Sentry instance]/saml/metadata/sentry/
>     Reply URL (Assertion Consumer Service URL): https//[URL of your Sentry instance]/saml/acs/sentry/
>     Sign on URL: https//[URL of your Sentry instance]/auth/login/sentry/
>     Relay State: https//[URL of your Sentry instance]/
>     Logout URL: https//[URL of your Sentry instance]/saml/sls/sentry/

Reviewer: ```suggestion
    Identifier (entity ID): https://sentry.io/saml/metadata/sentry/
    Reply URL (Assertion Consumer Service URL): https://sentry.io/saml/acs/sentry/
    Sign on URL: https://sentry.io/auth/login/sentry/
    Relay State: https://sentry.io/
    Logout URL: https://sentry.io/saml/sls/sentry/
```

    
8. In Section (3) labeled as `SAML Signing Certificate`, copy the `App Federation Metadata URL`

[ screenshot here ]

> 9. Go to `https://<YOUR_SENTRY_INSTANCE>/settings/sentry/auth/` and click on "Configure" 

Reviewer: ```suggestion
9. Go to `https://sentry.io/settings/<YOUR-ORGANIZATION-SLUG>/auth/` and click on "Configure" 
```

for Active Directory

10. Paste the App Federation Metadata URL and click "Get metadata"

11. In the following page, enter these to map the values from AzureAD to Sentry

>     ```jsx

Reviewer: ```suggestion
    ```
```

    IdP User ID: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name
    User Email: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress
    First Name: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname
    Last Name: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname
    ```

> 

Reviewer: It would be good to add in a link to https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/technical-reference/the-role-of-claims so our users can be empowered to find mappings for their custom configs.

---
title: 'Azure Active Directory SSO'
sidebar_order: 1
---

## Installation

> 1. Login to the [Azure portal](https://portal.azure.com/).

Reviewer 2: nit: Log in is the verb form :)

Author: Nice catch :)

> 2. (You can skip to (3) if you don't need to create a permission group for Sentry.) Under 

Reviewer 2: Suggest: If you don't require a permission group for Sentry, skip to Step 3.

Manage Azure Active Directory (the shield picture), click "View", which takes
you to the Organization Overview page. On the left hand navigation, click "Groups". Create a 
new group, add an owner, and members to the group. Set "Group type" to Office 365. For more 
details about group creation, see the
[Azure docs](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-groups-create-azure-portal).

> 3. On the left hand navigation, click "Groups". Create a new group, add an owner, and

Reviewer 2: do you also complete step 3 if you have completed step 2? (asking because they seem redundant)

members to the group. Set "Group type" to Office 365. For more details about group creation, see the
[Azure docs](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-groups-create-azure-portal).

4. Return to the Overview page. On the left hand navigation, click "Enterprise Applications".

> Then, select "Non-gallery application" and create one for Sentry. Once created, you'll

Reviewer 2: "create one for Sentry" is confusing - we should be explicit about what is being created.

be directed to "Sentry - Overview."

5. Click on `1. Assign users and groups`, and assign yourself and the group you've created 
to the Sentry app.

6. Click on `2. Setup Single Sign-on`

7. For Section (1) labeled as `Basic SAML2 Configuration`, enter the following data 
in each line.

    

>     Identifier (entity ID): https://sentry.io/saml/metadata/sentry/

Reviewer 2: should these be formatted as code? Example - Identifier (entity ID): `https://sentry.io/saml/metadata/sentry/`?

    Reply URL (Assertion Consumer Service URL): https://sentry.io/saml/acs/sentry/

    Sign on URL: https://sentry.io/auth/login/sentry/

    Relay State: https://sentry.io/

    Logout URL: https://sentry.io/saml/sls/sentry/
    
8. In Section (3) labeled as `SAML Signing Certificate`, copy the `App Federation Metadata URL`

[ screenshot here ]

9. Go to `https://sentry.io/settings/<YOUR-ORGANIZATION-SLUG>/auth/` and click on "Configure" 
for Active Directory

10. Paste the App Federation Metadata URL and click "Get metadata"

11. In the following page, enter these to map the values from AzureAD to Sentry.

    

>     IdP User ID: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name

Reviewer 2: code here as well for the entries?

    
    User Email: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress
    
    First Name: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname
    
    Last Name: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname
    
For more details about mappings for custom configs, see 
[The Role of Claims](https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/technical-reference/the-role-of-claims).

> 12. Congrats. You've set up an intance of AzureAD with Sentry.

Reviewer 2: This isn't a step. Keep it as plain text (or remove it if it's not consistent with our other Integrations doc).
